src/targets/targets.controller.ts

```diff
@@ -17,14 +17,22 @@ export class TargetsController {
     @Request() { user },
     @Body() { title, radius, latitude, longitude, topicId }: CreateTargetDto,
   ) {
-    const target = await this.targetservice.create(
+    return await this.targetservice.create(
       title,
       radius,
       latitude,
       longitude,
       user,
       topicId,
     )
-    return target
+  }
+
+  @UseGuards(AuthGuard())
+  @Get()
+  async index(
+    @Request() { user },
+  ){
+    const targets = await this.targetservice.findByUser(user)
+    return targets.map(target => new TargetIndexDto(target))
```

Comment:
The data mapper I was talking about should be called here, mapping dtos by hand is a bitch

Reply:
Ohh i'll check!